### ISSUE TITLE
Enable including data-files from eggs

### ISSUE THREAD

**Issue by htgoebel** [pyinstaller/pyinstaller#1362]
In Python 2.1, zipped eggs where included as zipped file, thus all package data was included, too. But modulegraph now only returns the modules imported from there.

We could now either
- change PyInstaller to include the the zipped egg-file, or
- make PyInstaller include the egg-data files somehow.
I'd prefer the later, since this would implement support for unzipped eggs, too (which are not supported in PyInstaller 2.1).

This this we need to investigate, how both [`pkgutil`](https://docs.python.org/3/library/pkgutil.html) and setuptool's [`pkg_resources`](https://pythonhosted.org/setuptools/pkg_resources.html) need this.

**Comment by htgoebel** [comment]
Todo; Check how py2exe, py2app and bbfreeze handle this. Are they including all data-files?

**Comment by htgoebel** [comment]
For the records (as this is blocking #1417 and #1402): I'm working on it and already have some idea how to to, but I'm short in spare time to spend on this ATM.

**Comment by htgoebel** [comment]
Recherche-Results:
- `py2exe` has no support for collecting data at all (which is documented)
- `bbfreeze` seams to rely on `pkg_resources.working_set `and `Distirbution`s to fetch some data. It says to copy egg-info directories. The code is in [`bbfreeze.freezer.EggAnalyzer`](https://github.com/schmir/bbfreeze/blob/c74223318826ebd1a19aed15cba5641055ff0187/bbfreeze/freezer.py#L56), which may give some hints.
- `py2app` seams copy all module data-files, but I did not find this to be documented.

**Comment by htgoebel** [comment]
PyInstaller 2.1 does:
- for a zipped egg: copy the egg into the distribution
- for unzipped-eggs: unhandled
- hooks are adding datafiles to the distribution-dirctory, not to the PYZ.

**Comment by matysek** [comment]
Is this a duplicate of #183 ?

**Comment by matysek** [comment]
`test_libraries.py::test_pylint` fails because `pkg_resources`   'Distribution' is not working.

**Comment by htgoebel** [comment]
> Is this a duplicate of #183?

No. #183 is about more magic, while this one is about restoring PyI 2.1 capabilities.

**Comment by htgoebel** [comment]
> test_libraries.py::test_pylint fails because pkg_resources 'Distribution' is not working.

???

**Comment by matysek** [comment]
> test_libraries.py::test_pylint fails because pkg_resources 'Distribution' is not working.
> ???

Then this comment is related to #183:  pylint depends on logilab-common and pylint uses pkg_resources to get version of logilab-common.

**Comment by htgoebel** [comment]
`pkgutil` and `pkg_resources` both just need a PEP 302-complinant *loader* supporting `get_data()`.

I made considerable progress on this: I implemented

- adding data-file into the PYZ
- automatically collecting the data-file from eggs (currently `.egg` must be in the name)
- `FrozenImporter.get_data()` accessing data contained in the PYZ.
- accessing the data ia `pkgutil.get_data()` and `pkg_resources.get_data()`

This could easily be extended to

1. automatically collect data-files from any type of Distribution not only eggs, this we may be able to drop quite some hooks
   The issue here is to find the Distribution, a given module-path belongs to. But solving this should be feasible.
2. include the meta-data (aka `EGG-INFO) into the PYZ.
  The issue here is that all Distributions store their meta-data in the `EGG_INFO` sub-directory. We need to find a way to distinguish the distribution, maybe by adding another directory-level. But where to get this level-name from?
